### PR TITLE
FilterForm availableGranularities optional

### PR DESCRIPTION
**Pull request by plavemar**
- marked availableGranularities for filterForm as optional

JIRA: RAIL-1842

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by plavemar** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3433/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5071/ : SUCCESS in 12m 28s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/872/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3434/ : SUCCESS in 1m 06s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/872/ : SUCCESS in 19m 15s

**Comment by plavemar** [comment]
extended check sonar

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR1307

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3435/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3/ : SUCCESS in 16m 24s (non-voting)

**Comment by plavemar** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3438/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5074/ : FAILURE in 4m 52s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3439/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/873/ : FAILURE in 5m 25s

File: examples/sdk-examples/ci/results/eslint-results.xml [hunk 1]
```diff
@@ -0,0 +1 @@
...ource="eslint.rules.@typescript-eslint/explicit-module-boundary-types" /><error line="33" column="75" severity="warning" message="Missing return type on function. (@typescript-eslint/explicit-module-boundary-types)" source="eslint.rules.@typescript-eslint/explicit-module-boundary-types" /><error line="78" column="33" severity="warning" message="Missing return type on function. (@typescript-eslint/explicit-module-boundary-types)" source="eslint.rules.@typescript-eslint/explicit-module-boundary-types" /><error line="78" column="40" severity="warning" message="Argument &apos;selector&apos; should be typed. (@typescript-eslint/explicit-module-boundary-types)" source="eslint.rules.@typescript-eslint/explicit-module-boundary-types" /><error line="78" column="50" severity="warning" message="Argument &apos;t&apos; should be typed. (@typescript-eslint/explicit-module-boundary-types)" source="eslint.rules.@typescript-eslint/explicit-module-boundary-types" /><error line="102" column="60" severity="warning" message="Missing return type on function. (@typescript-eslint/explicit-module-boundary-types)" source="eslint.rules.@typescript-eslint/explicit-module-boundary-types" /></file></checkstyle>
```
Comment (no23reason) [inline examples/sdk-examples/ci/results/eslint-results.xml:1]: Please remove this file

**Comment by plavemar** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3445/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5081/ : FAILURE in 5m 42s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/874/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3446/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/874/ : SUCCESS in 15m 20s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3448/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5084/ : FAILURE in 12m 08s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3451/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5087/ : SUCCESS in 12m 09s

**Comment by plavemar** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/875/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3452/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/875/ : SUCCESS in 15m 21s

**Comment by plavemar** [comment]
extended check sonar

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR1307

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3453/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/6/ : SUCCESS in 12m 56s (non-voting)

**Comment by plavemar** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/876/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3455/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/876/ : UNSTABLE in 15m 40s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3454/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5089/ : SUCCESS in 14m 21s

**Comment by plavemar** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/879/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3459/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/879/ : SUCCESS in 19m 43s

File: libs/sdk-backend-tiger/src/backend/workspace/dateFilterConfigs/index.ts [hunk 2]
```diff
@@ -24,8 +24,6 @@ const DefaultDateFilterConfig: IDateFilterConfig = {
     },
     relativeForm: {
         type: "relativeForm",
-        // month has to be the first as it should be the default selected option
-        availableGranularities: ["GDC.time.month", "GDC.time.date", "GDC.time.quarter", "GDC.time.year"],
```
Comment (no23reason) [inline libs/sdk-backend-tiger/src/backend/workspace/dateFilterConfigs/index.ts:28]: So what will this config do now? Will there still be the granularities available? Or do we need to move this to the "global" property of this type?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by plavemar** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3465/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5099/ : FAILURE in 12m 49s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/880/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3466/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/880/ : UNSTABLE in 16m 35s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3467/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5100/ : SUCCESS in 11m 58s

**Comment by plavemar** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/881/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by plavemar** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/882/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3470/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/882/ : SUCCESS in 15m 50s

File: libs/sdk-ui-filters/src/DateFilter/interfaces/index.ts [hunk 3]
```diff
@@ -30,11 +31,15 @@ export interface IUiAbsoluteDateFilterForm extends IAbsoluteDateFilterForm {
     to?: DateString;
 }
 
+export const isUiRelativeDateFilterForm = (obj: unknown): obj is IUiRelativeDateFilterForm => {
```
Comment (no23reason) [inline libs/sdk-ui-filters/src/DateFilter/interfaces/index.ts:34]: Please add a TSDoc and a release tag `@beta`. Also please make sure this function is exported by the `sdk-ui-filters` package.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3471/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5103/ : FAILURE in 3m 11s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3472/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5104/ : SUCCESS in 11m 58s

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1307

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1029/ : SUCCESS in 54s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1299/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/12/ : SUCCESS in 11m 41s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5105/ : SUCCESS in 12m 16s

**Pull request merged**